baseball/src/App.test.js

```diff
@@ -0,0 +1,67 @@
+
+//IMPORTS
+import React from 'react';
+import ReactDom from 'react-dom';
+import App from './App';
+import 'jest-dom/extend-expect';
+import { render , fireEvent } from 'react-testing-library';
+import Score from './components/score';
+import Megatron from './components/megatron';
+
+//BASEBALL TESTER SETUP ⬇︎
+it ( 'renders without crashing' , () => {
+    render ( <App/> )
+});
+
+describe ( 'The App Component' , () => {
+
+    const app = render ( <App/> );
+    const scoreComponent = render ( <Score/> );
+    const megatronComponent = render ( <Megatron/> );
+
+    it ( 'it exists' , () => {
+        expect ( app ) .toBeTruthy();
+    })
+    it ( 'contains content' , () => {
+        expect.anything();
+    })
+
+
+    const balls = app.getByText ( /balls/i );
+    const strikes = app.getByText ( /strikes/i );
+
+    describe ( 'Balls and Strikes' , () => {
+        it ( 'Balls' , () => {
+            expect ( balls ) .toHaveTextContent ( /balls/i )
+        })
+        it ( 'Strikes' , () => {
+            expect ( strikes ) .toHaveTextContent ( /strikes/i )
+        })
+    })
+
+    describe ( 'Action Buttons' , () => {
+        //BALLS ⬇︎
+        test ( 'Ball Button' , () => {
+            const ballButton = megatronComponent .getByTitle ( 'ballButton' )
+                fireEvent.click( ballButton );
+                const ballCount = scoreComponent .getByTitle ( 'ballCount' )
+                expect ( ballCount ) .toHaveTextContent( '1' );
+        })
+        //STRIKES ⬇︎
+        test ( 'Strike Button' , () => {
+            const strikeButton = megatronComponent .getByTitle ( 'strikeButton' )
+                fireEvent.click( strikeButton );
+                const strikeCount = scoreComponent .getByTitle ( 'strikeCount' )
+                expect ( strikeCount ) .toHaveTextContent( '1' );
+        })
+        //RESET SCORE ⬇︎
+        test ( 'Score resets at 3' , () => {
+            const strikeButton = megatronComponent .getByTitle ( 'strikeButton' )
+                fireEvent.click( strikeButton );
+                //using 1 strike button here only returned 2 strikes so i had to invoke it again to recieve 3 strikes so the counter would reset
+                fireEvent.click( strikeButton );
```

Comment:
note that the previous test is why the strike count is already at 1 when you fire the 2 events here. if you use the cleanUp method between tests, you'll need to alter this test to actually fire the 3 events